instrumentation/opentelemetry-instrumentation-urllib3/tests/test_urllib3_integration.py

```diff
@@ -309,3 +310,12 @@ def request_hook(span, request, headers, body):
         )
         self.assertIn("request_hook_body", span.attributes)
         self.assertEqual(span.attributes["request_hook_body"], body)
+
+    def test_request_body_stream(self):
+        URLLib3Instrumentor().uninstrument()
+        URLLib3Instrumentor().instrument()
+        body = BytesIO(b"123")
+
+        pool = urllib3.HTTPConnectionPool("httpbin.org")
+        response = pool.request("POST", "/status/200", body=body)
+        self.assertEqual(b"Hello!", response.data)
```

Comment:
How is this testing the fix? am I missing something?

Reply:
The assert itself is not really useful, but the test fail without the fix (Raise an error because we try to call len on a BytesIO).

Reply:
I didn't notice. Well, in that case, we would have to use something like `getbuffer.nbytes` for the size?

Reply:
I did a quick refactoring of the tests and added a special case for `io.BytesIO`.

Reply:
Consider using the format of the tests like [this](https://github.com/open-telemetry/opentelemetry-python-contrib/blob/7acc336ccac3df54b70452577365f8fcdfc80276/instrumentation/opentelemetry-instrumentation-tornado/tests/test_metrics_instrumentation.py#L108)
I think that it's more readable and clear what the test verifies each line, and easier to edit in the future.

Reply:
Done, I moved all the metrics test into their own test file.